### PR TITLE
Update Snapshot and Admin Panel Actions for Allocations

### PR DESCRIPTION
**Pull request by amit4111989**
…or allocations

## Description

Periodic update_snapshot function for allocations

## Checklist before merging Pull Requests
- [x ] New test(s) included to reproduce the bug/verify the feature
- [ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature
- [x] Reviewed and approved by at least one other contributor.
- [ ] If necessary, include a snippet in CHANGELOG.md

### CODE REVIEW

File: core/admin.py [hunk 1]
```diff
@@ -83,28 +88,60 @@ class AllocationSourceAdmin(admin.ModelAdmin):
...
+                api = AllocationSourceViewSet()
+                api.update(request, obj.uuid)
+
+            # if allocation source is end dated
+            if form.initial['end_date'] != obj.end_date:
+                api = AllocationSourceViewSet()
+                api.perform_destroy(obj, request=request)
+
+    def has_delete_permission(self, request, obj=None):
+        return False
```
Comment (steve-gregory) [inline core/admin.py:119]: This is an interesting way of doing "Django-admin interactions using established API views/serializers".

We could probably generalize this a little further and get greater utility (In a future branch).

File: cyverse_allocation/tasks.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+)
+from core.models import EventTable
+from core.models.allocation_source import total_usage
+from cyverse_allocation.cyverse_rules_engine_setup import CyverseTestRenewalVariables, CyverseTestRenewalActions, \
+    cyverse_rules
+
+logger = logging.getLogger(__name__)
+
+@task(name="update_snapshot_cyverse")
+def update_cyverse_snapshot(start_date=None, end_date=None):
```
Comment (steve-gregory) [inline cyverse_allocation/tasks.py]: Names do not match here

**Review by steve-gregory — COMMENTED** [review]
Comments below

File: cyverse_allocation/tasks.py (hunk 2 above)
Comment (amit4111989) [reply cyverse_allocation/tasks.py]: Thanks for pointing out steve, I pushed a commit which corrects this

File: core/admin.py (hunk 1 above)
Comment (amit4111989) [reply core/admin.py:119]: Yea I agree, we have never used API views/serializers from within the backend ( or have we ? ) so I wasn't sure how to go about it. Using requests would have been more cumbersome, but an abstraction layer which deals with this sort of situation would be perfect in the future

**Pull request merged**